File: app/src/main/java/khorunzhyicom/alex/theposter/presentation/activity/PosterController.java
package khorunzhyicom.alex.theposter.presentation.activity;


import android.support.annotation.NonNull;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.Arrays;

import butterknife.BindView;
import khorunzhyicom.alex.theposter.R;
import khorunzhyicom.alex.theposter.presentation.activity.adapter.PosterTabLayout;
import khorunzhyicom.alex.theposter.presentation.activity.adapter.PosterTabsAdapter;
import khorunzhyicom.alex.theposter.presentation.common.controller.ButterKnifeController;
import khorunzhyicom.alex.theposter.presentation.popular.PopularMoviesController;
import khorunzhyicom.alex.theposter.presentation.top.TopMoviesController;

public class PosterController extends ButterKnifeController<PosterView, PosterPresenter> implements PosterView {

    @BindView(R.id.poster_view_pager) ViewPager viewPager;

    @Override
    protected void onViewBound(View view) {
        super.onViewBound(view);
        viewPager.setAdapter(new PosterTabsAdapter(this, Arrays.asList(new TopMoviesController(), new PopularMoviesController())));
        ((PosterTabLayout)getActivity().findViewById(R.id.poster_tab_layout)).setViewPager(viewPager);
    }

    @Override
    protected View inflateView(@NonNull LayoutInflater inflater, @NonNull ViewGroup container) {
        return inflater.inflate(R.layout.controller_poster_home, container, false);
    }

    @NonNull
    @Override
    public PosterPresenter createPresenter() {
        return new PosterPresenter();
    }
}

File: app/src/main/java/khorunzhyicom/alex/theposter/presentation/activity/PosterController.kt
package khorunzhyicom.alex.theposter.presentation.activity


import android.os.Bundle
import android.support.v4.view.ViewPager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import findView
import khorunzhyicom.alex.theposter.App
import khorunzhyicom.alex.theposter.R
import khorunzhyicom.alex.theposter.di.components.injector.PresentationInjector
import khorunzhyicom.alex.theposter.presentation.activity.adapter.PosterTabLayout
import khorunzhyicom.alex.theposter.presentation.activity.adapter.PosterTabsAdapter
import khorunzhyicom.alex.theposter.presentation.activity.adapter.TabView
import khorunzhyicom.alex.theposter.presentation.common.controller.binder.ViewBinderController
import khorunzhyicom.alex.theposter.presentation.now.NowPlayingMoviesController
import khorunzhyicom.alex.theposter.presentation.popular.PopularMoviesController
import khorunzhyicom.alex.theposter.presentation.top.TopMoviesController
import khorunzhyicom.alex.theposter.presentation.upcoming.UpcomingMoviesController

class PosterController(args: Bundle? = null) : ViewBinderController<PosterView, PosterPresenter>(args), PosterView {

    val pager: ViewPager by findView<ViewPager>(R.id.poster_view_pager)

    override fun inflateView(inflater: LayoutInflater, container: ViewGroup): View {
        return inflater.inflate(R.layout.controller_poster_home, container, false)
    }

    override fun createPresenter(): PosterPresenter {
        return PosterPresenter()
    }

    override fun injector(): PresentationInjector = (activity as PosterActivity).injector()

    override fun onAttach(view: View) {
        super.onAttach(view)
        pager.adapter = PosterTabsAdapter(this, view.context, tabs())
        pager.offscreenPageLimit = pager.adapter.count
        (activity!!.findViewById(R.id.poster_tab_layout) as PosterTabLayout).setViewPager(pager)
    }

    override fun onDetach(view: View) {
        super.onDetach(view)
        pager.adapter = null
    }

    fun tabs(): List<TabView> {
        return listOf(NowPlayingMoviesController(),
                PopularMoviesController(),
                TopMoviesController(),
                UpcomingMoviesController())
    }
}
